File: src/main/java/com/example/sber/model/dto/response/StatusDto.java
package com.example.sber.model.dto.response;

import com.example.sber.model.enums.CurrentLevel;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class StatusDto {
    private CurrentLevel currentLevel;
    private int currentPoints;
    private int pointsToNext;
    private int progressPercent;
    private CurrentLevel nextLevel;

    public StatusDto (CurrentLevel currentLevel, int currentPoints, int pointsToNext, int progressPercent, CurrentLevel nextLevel) {
        this.currentLevel = currentLevel;
        this.currentPoints = currentPoints;
        this.pointsToNext = pointsToNext;
        this.progressPercent = progressPercent;
        this.nextLevel = nextLevel;
    }

    public StatusDto () {}
}

File: Sber_backend_kotlin/app/src/main/java/com/example/myapplication/data/model/StatusDto.kt
package com.example.myapplication.data.model

import com.google.gson.annotations.SerializedName

data class StatusDto(
    @SerializedName("currentLevel") val currentLevel: String,
    @SerializedName("currentPoints") val currentPoints: Int,
    @SerializedName("pointsToNext") val pointsToNext: Int,
    @SerializedName("progressPercent") val progressPercent: Int,
    @SerializedName("nextLevel") val nextLevel: String?
)
